File: baekjoon/silver/1024/1024.py
# input
# 1000000000 2
# output
# 199999998 199999999 200000000 200000001 200000002

def solution():
  n, k = map(int, input().split())
  while k <= 100:
    for i in range(int(n / k - k), int(n / k + k)):
      if i < 0:
        continue
      if k * (2 * i + (k - 1)) / 2 == float(n):
        return i, k
    k += 1
  return -1, 1
    
s, k = solution()
ans = [str(i) for i in range(s, s + k)]
print(' '.join(ans))

File: baekjoon/silver/1024/1024.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var n, k int

	fmt.Scanf("%d %d", &n, &k)
	s, d := solution(n, k)

	ans := make([]string, d)
	for i := 0; i < d; i++ {
		ans[i] = strconv.Itoa(s + i)
	}
	fmt.Println(strings.Join(ans, " "))
}

func solution(n, k int) (s, d int) {
	for ; k <= 100; k++ {
		for i := n/k - k; i < n/k+k; i++ {
			if i < 0 {
				continue
			}
			if float64(k)*(2*float64(i)+float64(k)-1)/2 == float64(n) {
				return i, k
			}
		}
	}
	return -1, 1
}
